### DISCUSSION TITLE
ZERO-SHOT Classification On SUN397

### DISCUSSION THREAD

**Discussion by xiaohoua** [mlfoundations/open_clip#1006] [Q&A] [answered]
I load ViT-H-14-378-quickgelu model  [huggingface](https://huggingface.co/apple/DFN5B-CLIP-ViT-H-14-378),and try to do zero-shot classification on SUN397, but i get very low accurancy.
So I want to know how openclip does classification on SUN397。
I think the reasons for my mistake may be as follows：
1.I downloaded the model weights and categorized them on the data set without fine-tuning training. Maybe I need to do some fine-tuning? (I tested 50 images from each category.)
2.The text prompt I used was probably wrong, I used the image category name as the text prompt, as shown in the image below:
![image](https://github.com/user-attachments/assets/aaf3fe58-8deb-4389-8b26-d62548a62138)
How do I do the classification code as [predict_accurancy.py](https://github.com/xiaohoua/clip-distillation/blob/branch-for-clip-SUN/SUN/predict_accurancy.py)
Did openclip open source the code and details of the experiments on the SUN dataset? Where can I get the information?
 Looking forward to reply!

**Comment by xiaohoua** [comment, marked as answer]
I get right result by [CLIP_benchmark](https://github.com/LAION-AI/CLIP_benchmark)

**Reply by rwightman** [reply]
this is the main source I know with open source / config for reproducing various benchmarks for these models, though some of the original openai evals are still hard to reproduce exactly due to some missing split specifications
